packages/client/hmi-client/src/components/home/tera-project-table.vue

```diff
@@ -1,23 +1,51 @@
 <template>
-	<DataTable :value="projects" dataKey="id" :rowsPerPageOptions="[10, 20, 50]" scrollable scrollHeight="45rem">
+	<!--The pt wrapper styling enables the table header to stick with the project search bar-->
+	<DataTable
+		ref="projectTableRef"
+		:value="projectsWithKnnMatches"
+		:rows="numberOfRows"
+		:rows-per-page-options="[10, 20, 30, 40, 50]"
+		:pt="{ wrapper: { style: { overflow: 'none' } } }"
+		data-key="id"
+		paginator
+		@page="getProjectAssets"
+	>
 		<Column
 			v-for="(col, index) in selectedColumns"
 			:field="col.field"
 			:header="col.header"
 			:sortable="!['stats', 'score'].includes(col.field)"
 			:key="index"
-			:style="`width: ${getColumnWidth(col.field)}%`"
 		>
 			<template #body="{ data }">
 				<template v-if="col.field === 'score'">
 					{{ Math.round((data.metadata?.score ?? 0) * 100) + '%' }}
 				</template>
 				<template v-if="col.field === 'name'">
-					<a class="project-title-link" @click.stop="emit('open-project', data.id)">
-						{{ data.name }}
-					</a>
+					<a @click.stop="emit('open-project', data.id)">{{ data.name }}</a>
+					<ul>
+						<li
+							v-for="(asset, index) in data.projectAssets.slice(0, data.showMore ? data.projectAssets.length : 3)"
+							class="flex align-center gap-2"
+							:key="index"
+						>
```

Comment:
I dunno if the projectAssets list returned are dynamic or not, but you may want to consider key'ing with assetId instead of index as a safer option, since index does not change even if the array-size or array-elements have changed.